core/include/traccc/geometry/detector.hpp

```diff
@@ -7,6 +7,9 @@
 
 #pragma once
 
+// Project include(s).
+#include "traccc/definitions/primitives.hpp"
```

Comment:
Was this needed here? I'm looking at the file, but can't see it. 😕

Reply:
I did not understand that either, but without, the toy detector benchmarks were not finding the arithmetic operators for vectors

Reply:
Doesn't that mean then that the benchmark needs this include? :confused: The code in this file really doesn't seem to require those definitions. :thinking:

Reply:
That's what I thought as well, but the operators were not found in detray code that the benchmarks depended on. I tried a few things, including defining the operators again in the benchmark files, but in the end it depended on the include ordering of this file into the benchmark header (before or after the other traccc includes), unless I included ```primitives``` here. Having this depend on the include order seemed more worrying to me than adding this include.
Let me try again, after the dust has settled now in detray. I agree that something is/was definitely fishy with how we have been using these operators so far.

If we go to the templated detray code, should this file contain fully defined types or templated alias types? I.e.
```
/// Default detector (also used for ODD)
template<typename algebra_t>
using default_detector = detector<detray::default_metadata<algebra_t>>;
```
because for fully defined types, we will need this include anyway, right?

Reply:
Seems to work now without! All the better